### ISSUE TITLE
Remove `ChainExpression` and `ChainElement`?

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#6371]
`foo?.()` is represented in our AST as (slightly abbreviated):

```
Expression::ChainExpression(
    ChainExpression {
        expression: ChainElement::CallExpression(
            CallExpression {
                callee: IdentifierReference { name: "foo" },
                arguments: [],
                optional: true,
            }
        )
    }
)
```

What do `ChainExpression` and `ChainElement` add here? They appear to just duplicate the same information that `optional` field holds.

Couldn't it just be:

```
Expression::CallExpression(
    CallExpression {
        callee: IdentifierReference { name: "foo" },
        arguments: [],
        optional: true,
    }
)
```

Or, if we want to follow Babel and shave 8 bytes off `CallExpression`, remove the `optional` field and use a separate enum variant:

```
Expression::OptionalCallExpression(
    CallExpression {
        callee: IdentifierReference { name: "foo" },
        arguments: [],
        // No `optional` field - `Expression::OptionalCallExpression` performs same function
    }
)
```

Same goes for the `MemberExpression` variants of `ChainElement` - they all also have an `optional` field.

Note: ESTree has `ChainExpression` and `ChainElement`. But I don't see the advantage of that design.

**Comment by Boshen** [comment]
No, AST nodes are defined by the spec. In this case https://tc39.es/ecma262/#sec-optional-chains

**Comment by Boshen** [comment]
`ChainExpression` is also really useful when we need to transform the syntax.

**Comment by Boshen** [comment]
> It's in the spec, but what does it actually do? What syntax cannot be expressed in AST without it?

Yes you can bend the AST to make it terser or more verbose ... but we don't design the AST, the spec does.

For runtimes, it needs to strictly follow the runtime semantics defined in https://tc39.es/ecma262/#sec-optional-chaining-chain-evaluation
